hazelcast/src/main/java/com/hazelcast/internal/partition/operation/AbstractPartitionOperation.java

```diff
@@ -129,4 +246,11 @@
     public final int getFactoryId() {
         return PartitionDataSerializerHook.F_ID;
     }
+
+    // return a new thread-safe Set populated with all elements from previouis
+    Set<Operation> newOperationSet(Collection<Operation> previous) {
```

Comment:
what was the reason of thread-safe-ness?

Reply:
With the new replication operation preparation happening either on partition or async threads, the collection of replication operations may be mutated from different threads.